synapse/util/caches/stream_change_cache.py

```diff
@@ -74,12 +74,17 @@ def get_entities_changed(self, entities, stream_pos):
         assert type(stream_pos) is int
 
         if stream_pos >= self._earliest_known_stream_pos:
-            result = {
+            changed_entities = {
                 self._cache[k] for k in self._cache.islice(
                     start=self._cache.bisect_right(stream_pos),
                 )
             }
 
+            result = {
```

Comment:
it'll be quicker to do `result = changed_entities.intersection(entities)` here, since `intersection` is an optimised bit of C (which does do the right thing when given a non-set argument)

Reply:
ooh, cool